documentation/book/ref-operators-cluster-operator-configuration.adoc

```diff
@@ -57,12 +57,14 @@ The required syntax is whitespace or comma separated `_<version>_=_<image>_` pai
 For example `{KafkaVersionLower}={DockerKafkaImagePrevious}, {KafkaVersionHigher}={DockerKafkaImageCurrent}`.
 This is used when a `KafkaConnect.spec.version` property is specified but not the `KafkaConnect.spec.image`, as described in xref:assembly-configuring-container-images-deployment-configuration-kafka-connect[].
 
-`STRIMZI_KAFKA_CONNECT_S2I_IMAGES`:: Required.
+ifdef::OpenShift[]
+`STRIMZI_KAFKA_CONNECT_S2I_IMAGES`:: Required on {OpenShiftName}
```

Comment:
Is this really required on OpenShift? Is it required on OpenShift only?

Reply:
From the PoV of the code, I think we should should require this on platforms with S2I support. That's so that users can just create `KafkaConnectS2I` resources with just a `spec.version` (and without a `spec.image`) and be guaranteed that it just work for the "supported versions" for the operator. But I've seen your comment on #1724, and I can agree that it's complicated to do this validation right now. But it's weird to pass this option for platforms which don't support it. And if we pass it, we ought to document it. But that means in pure-kube doc we just start talking about s2i out of nowhere. 

Because we're not planning on actually using pure-kube doc I'll just remove the `ifdef`. I'll also keep the Required on {OpenShiftName} to try to make clear what _should_ be provided on OpenShift, even if we don't actually validate it in the CO right now.

In the longer term, I wonder if we should provide different install resources for kubernetes and OpenShift. For example, there's no point in installing the S2I CRD on kubernetes. I know it complicated the build rather a lot, so it's not something I'm advocating currently.